contract/models/contract.py

```diff
@@ -209,7 +209,8 @@ def _onchange_contract_template_id(self):
                     field.name in self.NO_SYNC,
                 )
             ):
-                self[field_name] = self.contract_template_id[field_name]
+                if self.contract_template_id[field_name]:
+                    self[field_name] = self.contract_template_id[field_name]
```

Comment:
Maybe we need to play some on change here, e.g. `if field_name == "contract_type": self._onchange_contract_type()`.